### PR TITLE
fix: only use the export chunk for exported web components if it is found

### PR DESCRIPTION
**Pull request by Artur-**
Makes it possible to reconfigure webpack to put exported wcs elsewhere

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandler.java [hunk 1]
```diff
@@ -96,7 +97,14 @@ private WebComponentBootstrapContext(VaadinRequest request,
             extends BootstrapPageBuilder {
         @Override
         protected List<String> getChunkKeys(JsonObject chunks) {
-            return Collections.singletonList(EXPORT_CHUNK);
+            boolean hasExportChunk = Arrays.stream(chunks.keys())
+                    .anyMatch(s -> EXPORT_CHUNK.equals(s));
```
[the remark this answers is not in the document]
Comment (joheriks) [reply flow-server/src/main/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandler.java:101]: Or use `chunks.hasKey(EXPORT_CHUNK)`?

**Review by joheriks — DISMISSED** [review]

**Review by joheriks — DISMISSED** [review]

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]
Let's add some unit tests for this change before we merge it. The fusion team can make the tests.

**Comment by joheriks** [comment]
Rebased and added unit test.

File: flow-server/src/test/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandlerTest.java [hunk 2]
```diff
@@ -257,6 +258,39 @@ public void canHandleRequest_hasWebComponentConfigPathIsWebComponentUI_returnsTr
         Assert.assertTrue(handler.canHandleRequest(request));
     }
 
+    @Test
+    public void writeBootstrapPage_noExportChunk()
```
Comment (haijian-vaadin) [inline flow-server/src/test/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandlerTest.java:262]: Should we also have a test like `writeBootstrapPage_withExportChunk`?

**Review by haijian-vaadin — CHANGES_REQUESTED** [review]

File: flow-server/src/test/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandlerTest.java (hunk 2 above)
Comment (joheriks) [reply flow-server/src/test/java/com/vaadin/flow/server/communication/WebComponentBootstrapHandlerTest.java:294]: Good idea, done.

**Review by haijian-vaadin — APPROVED** [review]

**Pull request merged**